### PR TITLE
Reduce spacing at top of all pages

### LINKED ISSUE #1134: Reduce space at top of archive pages
Currently there is a large space at the top of archive pages which is reducing the content a user can see above the fold:

Screen Shot 2022-11-30 at 3 40 40 PM

This is caused the following CSS in `_legacy-styles.scss`, a file which is not in the repo:

```
/*--------------------------------------------------------------
# bbPress Specific
--------------------------------------------------------------*/
/* bbPress specific styles */
/* Many of these override the styles from the plugin */
section {
	padding: 4rem 0;
}
```

I think this value could be halved for the first `section` in the page.

### PR DESCRIPTION
**Pull request by adamwoodnz**
See #1134 

Includes removing course specific padding, for more consistency across site.

### Screenshots
| Page | Before | After |
|-|-|-|
| Tutorials | Screen Shot 2022-12-01 at 2 45 15 PM | Screen Shot 2022-12-01 at 2 43 15 PM |
| Online Workshops | Screen Shot 2022-12-01 at 2 44 25 PM | Screen Shot 2022-12-01 at 2 42 16 PM |
| Courses | Screen Shot 2022-12-01 at 2 46 17 PM | Screen Shot 2022-12-01 at 2 39 06 PM |
| Lesson Plans | Screen Shot 2022-12-01 at 2 47 08 PM | Screen Shot 2022-12-01 at 2 41 06 PM |

### How to test
Browse as many pages as possible, checking for consistent space above the page title, and for unusual gaps between blocks of content (adjacent `section` elements) further down the page.

#### Key pages
- Home (no changes)
- all landing pages
- Courses archive, single Course and Lesson
- search

### CODE REVIEW

File: wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss [hunk 1]
```diff
@@ -14,12 +14,17 @@ body:not(.trac):not(.home-page) #main {
 
 h1, h2, h3, h4 { color: #555 !important; }
 
-/* ___ Homepage ___ */
+section {
+	padding-top: 0;
 
-#page.home {
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss:19]: Markup matching this is never rendered

File: wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss [hunk 2]
```diff
@@ -5,7 +5,7 @@
 */
 
 body:not(.trac):not(.home-page) #main {
-	margin-top: 0.5rem;
+	margin-top: 2rem;
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss:8]: This is a more reliable way of controlling the space at the top of every page, as there isn't always a `section`, eg. on regular pages like Online Workshops

File: wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss [hunk 3]
```diff
@@ -14,12 +14,17 @@ body:not(.trac):not(.home-page) #main {
 
 h1, h2, h3, h4 { color: #555 !important; }
 
-/* ___ Homepage ___ */
+section {
+	padding-top: 0;
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss:18]: Remove the huge space at the top of standalone sections

File: wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss [hunk 4]
```diff
@@ -14,12 +14,17 @@ body:not(.trac):not(.home-page) #main {
...
 
-#page.home {
-	background-color: #fff;
+	+ section,
+	+ hr + section {
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss:21]: Add the large spacing back if this is a layout broken into sections, including where an hr is used between.

File: wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss [hunk 5]
```diff
@@ -30,13 +35,17 @@ h1, h2, h3, h4 { color: #555 !important; }
 	max-width: 650px;
 }
 
+.quick-intro {
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/vendor/_overrides.scss:38]: Special treatment for the coloured blocks on the home page

File: wp-content/themes/pub/wporg-learn-2020/css/vendor/_sensei.scss [hunk 6]
```diff
@@ -5,7 +5,6 @@
 		}
 	}
 
-	&.course:not(.post-type-archive-course) article.course:first-child,
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2020/css/vendor/_sensei.scss:8]: Not entirely sure what this is supposed to target, but came from the customiser and ends up breaking the first card in the search results

**Review by hlashbrooke — APPROVED** [review]
Looks good to me - the reduced spacing is much neater all round :shipit:

**Pull request merged**